### DISCUSSION TITLE
Set-GitHubContent binary / text problems

### DISCUSSION THREAD

**Discussion by turbo-donkey** [microsoft/PowerShellForGitHub#326] [Q&A]
I'm trying to push a list of IP's up to my repo but as soon as I use Set-GitHubContent on the target file it becomes a binary file in GitHub and its raw URL can no longer be processed by my firewall, is there some way to maintain the target file without it changing to binary?

(not massively experienced with GitHub so go easy on me!)

Set-GitHubContent -OwnerName My-Owner-Name -RepositoryName MyRepoName -Path targetfile.txt -CommitMessage 'Automated PoSH commit' -Content ($list_of_ips | Out-String) -Verbose

Pulling my hair out here!

I've been testing if the raw URL of the file can be rendered in browser or passes a file to download - rendered in browser means GitHub is treating it as text and my firewall can process no problem, if a download is passed its being treated as binary and my firewall wont look at it.

**Comment by HowardWolosky** [comment]
Let's try removing PowerShellForGitHub from the equation here for a moment.
Try both of these scenarios:
1. Use command-line `git.exe` to `git add myfile.txt` and then `git push` to your repo.  Then see if your firewall treats that as binary when retrieving.
2. Use GitHub.com's browser-UI to upload a text file and then test to see if your firewall treats that as a binary when retrieving.

My guess is that this has nothing to do with how this module interacts with the GitHub API, but rather what headers GItHub is setting when sending the content via URL requests.

**Comment by jorioux** [comment]
Fixed with this commit:
https://github.com/microsoft/PowerShellForGitHub/commit/1200b5b36aa94d96906802ddc04f8604ab83d83c
